star-rx72n-firmware/lib/rx_usb/src/rx_usb.c

```
@@ -197,25 +341,78 @@ STATIC_TESTABLE uint32_t internal_ring_buffer_read(ring_buffer_t* buf,
+    return;
+  }
+
   /* Only trigger if data is available */
-  if (internal_ring_buffer_available(&s_usb.tx_buffer) == k_min_transfer_size) {
+  if (internal_ring_buffer_available(&s_usb.ports[port].tx_buffer) == k_min_transfer_size) {
     return;
   }
 
-  /* Check if Pipe 1 is not busy (PBUSY bit = 0 means idle) */
-  if ((usb0()->pipe1ctr & k_usb_pipectr_pbusy) == k_ring_buffer_empty) {
+  const uint8_t pipe = s_port_configs[port].pipe_bulk_in;
```

Clang (14.0.6)

[warning] variable 'pipe' is not initialized

(cppcoreguidelines-init-variables)